iphone/cli/hooks/frameworks.js

```diff
@@ -740,20 +754,37 @@ class FrameworkInspector {
 	 * @access public
 	 */
 	async inspect(frameworkPath) {
-		if (frameworkPath.endsWith('.xcframework')) {
-			const frameworkName = path.basename(frameworkPath, '.xcframework');
-			// we actually should have 2+ frameworks underneath. So what do we do? Record each?
-			const frameworkInfo = new FrameworkInfo(frameworkName, frameworkPath);
-			this._logger.debug(`Found xcframework ${frameworkName.green} at ${frameworkPath.cyan}`);
-			return frameworkInfo;
-		}
-
-		const frameworkName = path.basename(frameworkPath, '.framework');
-		const binaryPathAndFilename = path.join(frameworkPath, frameworkName);
-		const result = await this.detectBinaryTypeAndArchitectures(binaryPathAndFilename);
-		const frameworkInfo = new FrameworkInfo(frameworkName, frameworkPath, result.type, result.architectures);
+		const packageExtension = path.extname(frameworkPath);
+		const frameworkName = path.basename(frameworkPath, packageExtension);
+		let frameworkInfo;
+		let type;
+		const meta = { name: frameworkName, path: frameworkPath };
+		if (packageExtension === '.framework') {
+			const binaryPathAndFilename = path.join(frameworkPath, frameworkName);
+			const result = await this.detectBinaryTypeAndArchitectures(binaryPathAndFilename);
+			meta.type = type = result.type;
+			meta.architectures = result.architectures;
+			frameworkInfo = new FrameworkInfo(meta);
+		} else {
+			const archs = new Set();
```

Comment:
I still think this code and most of our CLI code should move away from bare architecture sets like this, as they're rendered ambiguous by the new Apple Silicon stuff. `'arm64'` is now ambiguous and is not implicitly iOS device. So I'm not sure collecting this info is actually useful unless we're combining arch with "platform".

In fact, the move to xcframework makes our use of architectures in module `manifest`s redundant and useless since the xcframework spells out exactly the set of targets and architectures it holds.

Reply:
The collection of archs here is only done to determine if a third-party framework uses a fat binary that needs to be stripped of unused archs when doing a dist build. It's true that XCFramework packages don't need this anymore since Xcode makes sure to only link with the correct library archs, but i still decided to collect all archs present in the package to have the complete metadata, just in case.